client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -640,3 +642,76 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def get_export_preset_by_name(preset_name: str) -> substance_painter.export.Resource:
+    export_presets= get_export_presets()
+    preset_full_name = export_presets[preset_name]
+    for export_preset in substance_painter.export.list_resource_export_presets():
+        if export_preset.resource_id.name == preset_full_name:
+            return export_preset
+
+
+def get_export_preset_with_filtered_maps(
+        export_preset: substance_painter.export.Resource,
+        channel_type_names: list[str]) -> dict:
+    filtered_maps = []
+    for output_map in export_preset.list_output_maps():
+        output_filename = output_map.get("fileName")
+        if not output_filename:
+            continue
+
+        if any(
+            channel_type_name in output_filename
+            for channel_type_name in channel_type_names
+        ):
+            filtered_maps.append(output_map)
+    # Create a new preset
+    return {
+        "exportPresets": [
+            {
+                "name": export_preset.resource_id.name,
+                "maps": filtered_maps
+            }
+        ],
+    }
+
+
+@contextlib.contextmanager
+def set_layer_stack_opacity(node_ids, channel_types):
+    """Function to set the opacity of the layer stack during
+    context
+    Args:
+        node_ids (list): A list of substance painter node ids
+        channel_types (list): A list of channel types
+    """
+    # Do nothing
+    if not node_ids or not channel_types:
+        yield
+        return
+    all_selected_nodes = []
+    original_opacity_values = []
+    stack = substance_painter.textureset.get_active_stack()
+    stack_root_layers = (
+        substance_painter.layerstack.get_root_layer_nodes(stack)
+    )
+
+    for node_id in node_ids:
+        node = substance_painter.layerstack.get_node_by_uid(int(node_id))
+        all_selected_nodes.append(node)
+    filtered_nodes = {node for node in stack_root_layers
+                      if node not in all_selected_nodes}
```

Comment:
```suggestion
    stack = substance_painter.textureset.get_active_stack()
    stack_root_layers = (
        substance_painter.layerstack.get_root_layer_nodes(stack)
    )

    all_selected_nodes = []
    for node_id in node_ids:
        node = substance_painter.layerstack.get_node_by_uid(int(node_id))
        all_selected_nodes.append(node)
    filtered_nodes = {node for node in stack_root_layers
                      if node not in all_selected_nodes}
    original_opacity_values = []
```

I'm also wondering whether we should refactor `filtered_nodes` to `disabled_nodes` or `excluded_nodes` so that it's clear which of the two sides it's filtering to. Otherwise it may be ambiguous whether it filters to nodes that need to be enabled or disabled.